File: src/Circit.Weather.Api/Models/WeatherModel.cs
namespace Circit.Weather.Api.Models
{
    public class WeatherModel
    {
        public Location Location { get; set; }
        public WeatherCurrent Current { get; set; }
    }

    public class WeatherCurrent
    {
        public long Last_updated_epoch { get; set; }
        public string Last_updated { get; set; }
        public double Temp_c { get; set; }
        public double Temp_f { get; set; }
        public int Is_day { get; set; }
        public WeatherCondition Condition { get; set; }
        public double Wind_mph { get; set; }
        public double Wind_kph { get; set; }
        public int Wind_degree { get; set; }
        public string Wind_dir { get; set; }
        public double Pressure_mb { get; set; }
        public double Pressure_in { get; set; }
        public double Precip_mm { get; set; }
        public double Precip_in { get; set; }
        public int Humidity { get; set; }
        public int Cloud { get; set; }
        public double Feelslike_c { get; set; }
        public double Feelslike_f { get; set; }
        public double Vis_km { get; set; }
        public double Vis_miles { get; set; }
        public double Uv { get; set; }
        public double Gust_mph { get; set; }
        public double Gust_kph { get; set; }
    }

    public class WeatherCondition
    {
        public string Text { get; set; }
        public string Icon { get; set; }
        public int Code { get; set; }
    }
}

File: src/Circit.Weather.Web/src/app/models/weather-model.ts
export interface WeatherModel {
    Location: {
        Name: string;
        Region: string;
        Country: string;
        Lat: number;
        Lon: number;
        Tz_id: string;
        Localtime_epoch: number;
        Localtime: string;
    }
    Current: {
        Last_updated_epoch: number;
        Last_updated: string;
        Temp_c: number;
        Temp_f: number;
        Is_day: number;
        Condition: {
            Text: string;
            Icon: string;
            Code: number;
        }
        Wind_mph: number;
        Wind_kph: number;
        Wind_degree: number;
        Wind_dir: string;
        Pressure_mb: number;
        Pressure_in: number;
        Precip_mm: number;
        Precip_in: number;
        Humidity: number;
        Cloud: number;
        Feelslike_c: number;
        Feelslike_f: number;
        Vis_km: number;
        Vis_miles: number;
        Uv: number;
        Gust_mph: number;
        Gust_kph: number;   
    };
}
